src/v/kafka/server/quota_manager.cc

```diff
@@ -344,17 +344,13 @@ ss::future<> quota_manager::do_gc(clock::time_point expire_threshold) {
         return res;
     };
 
-    auto reducer = [](const key_set& acc, const key_set& next) -> key_set {
-        key_set result;
-        // Note: std::set_intersection assumes the inputs are sorted and
-        // guarantees that the output is sorted
-        std::set_intersection(
-          acc.begin(),
-          acc.end(),
-          next.begin(),
-          next.end(),
-          std::back_inserter(result));
-        return result;
+    auto reducer = [](key_set acc, const key_set& next) -> key_set {
+        // In-place set intersection assumes that the inputs are sorted and
+        // guarantees that the output is also sorted
+        auto it = std::set_intersection(
+          acc.begin(), acc.end(), next.begin(), next.end(), acc.begin());
+        acc.resize(it - acc.begin());
```

Comment:
You can use `chunked_vector::erase_to_end()`
```suggestion
        acc.erase_to_end(it);
```

Reply:
Done